### DISCUSSION TITLE
FEM problem error

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27334] [Q&A Modules: Phase field]
I am trying to create an input script for crystal plasticity and below is the error that occurs in the .out file. Need some help. Also are there simple deformation models/examples of crystal plasticity in the MOOSE repository?  I need some directions if there are. Below is the error message and my input script.

Error message:
Material property 'stress', requested by 'TM_TensorMechanics0' is not defined on block 0
Material property 'Jacobian_mult', requested by 'TM_TensorMechanics0' is not defined on block 0
Material property 'stress', requested by 'TM_TensorMechanics1' is not defined on block 0
Material property 'Jacobian_mult', requested by 'TM_TensorMechanics1' is not defined on block 0
Material property 'phase0_mechanical_strain', requested by 'crysp_phase0' is not defined on block 0
Material property 'crysrot', requested by 'crysp_phase0' is not defined on block 0
Material property 'crysrot', requested by 'slip_rate_gss_phase0' is not defined on block 0
Material property 'stress', requested by 'stress_yy' is not defined on block 0
Material property 'stress', requested by 'vonmises' is not defined on block 0

Input script:
[Mesh]
  [file]
    type = FileMeshGenerator
    file = kks_multiphase_largescale_large_out.e-s183
    use_for_exodus_restart = true
   []
[]
  
[Variables]
  [./disp_x]
  [../]

  [./disp_y]
  [../]

  # order parameter 0
  [./eta0]
    initial_from_file_var = eta1
  [../]
  
[]

  
[Kernels]
  [./eta0_dt]
    type = TimeDerivative
    variable = eta0
  [../]

  [./TensorMechanics]
    displacements = 'disp_x disp_y'
    incremental = true
  [../]
[]

  
[AuxVariables]
  [./stress_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./e_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  
  [./vonmises]
    order = CONSTANT
    family = MONOMIAL
  [../]

[]
  
[Functions]
  [./tdisp]
    type = ParsedFunction
    expression = 0.1*t
  [../]
[]
  
  
[AuxKernels]
  [./stress_yy]
    type = RankTwoAux
    block = 0
    variable = stress_yy
    rank_two_tensor = stress
    index_j = 1
    index_i = 1
    execute_on = timestep_end
    
  [../]
  [./e_yy]
    type = RankTwoAux
    variable = e_yy
    rank_two_tensor = lage
    index_j = 1
    index_i = 1
    execute_on = timestep_end
  [../]
  
  [./vonmises]
    type = RankTwoScalarAux
    block = 0
    rank_two_tensor = stress
    variable = vonmises
    scalar_type = VonMisesStress
    execute_on = timestep_end
  [../]
[]
  
[BCs]
  [./fix_x]
    type = DirichletBC
    variable = disp_x
    boundary = 'left'
    value = 0
  [../]
  [./fix_y]
    type = DirichletBC
    variable = disp_y
    boundary = 'bottom'
    value = 0
  [../]
  [./tdisp]
    type = FunctionDirichletBC
    variable = disp_y
    boundary = top
    function = tdisp
  [../]
[]
  
[UserObjects]
  [./slip_rate_gss_phase0]
    type = CrystalPlasticitySlipRateGSS
    variable_size = 12
    slip_sys_file_name = input_slip_sys.txt
    num_slip_sys_flowrate_props = 2
    flowprops = '1 4 1e-6 0.05 5 8 1e-6 0.05 9 12 1e-6 0.05'
    uo_state_var_name = state_var_gss_phase0
  [../]
  [./slip_resistance_gss_phase0]
    type = CrystalPlasticitySlipResistanceGSS
    variable_size = 12
    uo_state_var_name = state_var_gss_phase0
  [../]
  [./state_var_gss_phase0]
    type = CrystalPlasticityStateVariable
    variable_size = 12
    groups = '0 4 8 12'
    group_values =  '60.8 60.8 60.8'
    uo_state_var_evol_rate_comp_name = state_var_evol_rate_comp_voce_phase0
    scale_factor = 1.0
  [../]
  [./state_var_evol_rate_comp_voce_phase0]
    type = CrystalPlasticityStateVarRateComponentGSS
    variable_size = 12
    groups = '0 4 8 12'
    hprops = '1.0 541.5 109.8 2.5'
    uo_slip_rate_name = slip_rate_gss_phase0
    uo_state_var_name = state_var_gss_phase0
  [../]
[]
  
[Materials]
  [./crysp_phase0]
    type = FiniteStrainUObasedCP
    block = 0
    stol = 1e-2
    tan_mod_type = exact
    uo_slip_rates = 'slip_rate_gss_phase0'
    uo_slip_resistances = 'slip_resistance_gss_phase0'
    uo_state_vars = 'state_var_gss_phase0'
    uo_state_var_evol_rate_comps = 'state_var_evol_rate_comp_voce_phase0'
    base_name = phase0
  [../]
  [./strain_phase0]
    type = ComputeFiniteStrain
    displacements = 'disp_x disp_y'
    block = 0
  [../]
  [./elasticity_tensor_phase0]
    type = ComputeElasticityTensorCP
    C_ijkl = '1.684e5 1.214e5 1.214e5 1.684e5 1.214e5 1.684e5 0.754e5 0.754e5 0.754e5'
    fill_method = symmetric9
    base_name =phase0
    block = 0
  [../]
[]  
  
[Postprocessors]
  [./stress_yy]
    type = ElementAverageValue
    variable = stress_yy
  [../]
  [./e_yy]
    type = ElementAverageValue
    variable = e_yy
  [../]
  
  [./vonmises]
    type = ElementAverageValue
    variable = vonmises
    block = 'ANY_BLOCK_ID 0'
  [../]
[]
  
[Preconditioning]
  [./smp]
    type = SMP
    full = true
  [../]
[]
  
[Executioner]
  type = Transient
  solve_type = 'PJFNK'

  petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -snes_type'
  petsc_options_value = 'lu            superlu_dist          vinewtonrsls'  

#  petsc_options_iname = '-pc_type -pc_asm_overlap -sub_pc_type -ksp_type -ksp_gmres_restart'
#  petsc_options_value = ' asm      2              lu            gmres     200'
  l_max_its = 20
  nl_max_its = 10
  nl_rel_tol = 1.0e-8
  nl_abs_tol = 1.0e-9

  end_time = 300

  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 5e-3
    cutback_factor = 0.75
    growth_factor = 1.2
    optimal_iterations = 20
  [../]
[]
  
[Outputs]
  exodus = true
  csv = true
[]

**Comment by GiudGiud** [comment]
Hello

Use Debug/show_material_props to see the names of the material properties created and requested. Then make sure they match. 

It looks like a prefix is being used. This could be creating the mismatch in the names

Guillaume

**Reply by Emmanuel-324** [reply]
Below is the output after adding the [Debug] block
```
Active Materials:
    Subdomain:  (0)
      Material Name:       crysp_phase0
      Property Names:      "fp" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" "phase0_stress" 
                             "pk2" "slip_rate_gss_phase0" "slip_rate_gss_phase0_flow_direction" "slip_resistance_gss_phase0" 
                             "state_var_evol_rate_comp_voce_phase0" "state_var_gss_phase0" "update_rot" 
                             
      Material Name:       strain_phase0
      Property Names:      "deformation_gradient" "mechanical_strain" "rotation_increment" "strain_increment" 
                             "strain_rate" "total_strain" 
      Material Name:       elasticity_tensor_phase0
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             

    Active Face Materials:
    Subdomain:  (0)
      Material Name:       crysp_phase0_face
      Property Names:      "fp" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" "phase0_stress" 
                             "pk2" "slip_rate_gss_phase0" "slip_rate_gss_phase0_flow_direction" "slip_resistance_gss_phase0" 
                             "state_var_evol_rate_comp_voce_phase0" "state_var_gss_phase0" "update_rot" 
                             
      Material Name:       strain_phase0_face
      Property Names:      "deformation_gradient" "mechanical_strain" "rotation_increment" "strain_increment" 
                             "strain_rate" "total_strain" 
      Material Name:       elasticity_tensor_phase0_face
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             

Active Neighboring Materials:
    Subdomain:  (0)
      Material Name:       crysp_phase0_neighbor
      Property Names:      "fp" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" "phase0_stress" 
                             "pk2" "slip_rate_gss_phase0" "slip_rate_gss_phase0_flow_direction" "slip_resistance_gss_phase0" 
                             "state_var_evol_rate_comp_voce_phase0" "state_var_gss_phase0" "update_rot" 
                             
      Material Name:       strain_phase0_neighbor
      Property Names:      "deformation_gradient" "mechanical_strain" "rotation_increment" "strain_increment" 
                             "strain_rate" "total_strain" 
      Material Name:       elasticity_tensor_phase0_neighbor
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             

Active Boundary Materials:
                           ```

**Reply by GiudGiud** [reply]
So the materials properties you need have their names prefixed and/or suffixed by phase0 seemingly

You ll have to be mindful of that in the input
